### PR TITLE
Move Industry, Town, Company, Station into World folder

### PR DESCRIPTION
**Pull request by duncanspumpkin**

### CODE REVIEW

File: src/OpenLoco/CMakeLists.txt [hunk 1]
```diff
@@ -56,8 +56,8 @@ set(OLOCO_SOURCES
     "${CMAKE_CURRENT_SOURCE_DIR}/src/Audio/OpenAL.cpp"
     "${CMAKE_CURRENT_SOURCE_DIR}/src/Audio/VehicleChannel.cpp"
     "${CMAKE_CURRENT_SOURCE_DIR}/src/CommandLine.cpp"
-    "${CMAKE_CURRENT_SOURCE_DIR}/src/Company.cpp"
-    "${CMAKE_CURRENT_SOURCE_DIR}/src/CompanyManager.cpp"
+    "${CMAKE_CURRENT_SOURCE_DIR}/src/World/Company.cpp"
```
Comment (duncanspumpkin) [inline src/OpenLoco/CMakeLists.txt:59]: should probably sort these

Comment (AaronVanGeffen) [reply src/OpenLoco/CMakeLists.txt:59]: Yes please 🙂

**Comment by ZehMatt** [comment]
I'm honestly not quite sure this is the right thing to do, Engine should only have basic building blocks for a game, I thin TownManager IndustryManager etc seem all quite game specific if the Engine is to be re-used the next game has maybe no use for this.

**Review by ZehMatt — APPROVED** [review]
Looks good to me, needs the files sorted.

**Pull request merged**